internal/tjson/tjson.go

```diff
@@ -0,0 +1,217 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package tjson provides converters from/to JSON with JSON Schema for built-in and `types` types.
+//
+// See contributing guidelines and documentation for package `types` for details.
+package tjson
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+
+	"github.com/AlekSi/pointer"
+
+	"github.com/FerretDB/FerretDB/internal/types"
+	"github.com/FerretDB/FerretDB/internal/util/lazyerrors"
+)
+
+// tjsontype is a type that can be marshaled from/to JSON with JSON Schema.
+type tjsontype interface {
+	tjsontype() // seal for go-sumtype
+
+	json.Marshaler
+}
+
+//go-sumtype:decl tjsontype
+
+// checkConsumed returns error if decoder or reader have buffered or unread data.
```

Comment:
Do we have a comment about `checkConsumed` - that it's for fuzzing purposes somewhere? Can it be added here?